### ISSUE TITLE
UnnecessaryLet false negative

### ISSUE THREAD

**Issue by BraisGabin** [detekt/detekt#3786]
## Expected Behavior

```kotlin
value?.let {
  listOf().map { it }
}
```

Should be marked as an unnecessary usage of `let` because we aren't using `value` at all

## Observed Behavior
The code is not flagged. But this code is flagged correctly:

```kotlin
value?.let { _ ->
  listOf().map { it }
}
```

In the first exampe, it seems that the rule thinks that `it` is from `let` instead of `map` so it thinks that it's a legit usage of `let`.

## Your Environment
* Version of detekt used: 1.17.0
